pkg/execution/state/redis_state/backlog_normalization.go

```diff
@@ -455,7 +471,10 @@ func (q *queue) BacklogNormalizePeek(ctx context.Context, b *QueueBacklog, limit
 		isMillisecondPrecision: true,
 	}
 
-	res, err := p.peek(ctx, backlogSet, false, q.clock.Now(), limit)
+	// this is essentially +inf as no queue items should ever be scheduled >2y out
+	normalizeLookahead := q.clock.Now().Add(time.Hour * 24 * 365 * 2)
+
+	res, err := p.peek(ctx, backlogSet, false, normalizeLookahead, limit)
```

Comment:
This is the only real logic change. We switch from loading items up until now -> peeking a random offset of _all_ items in the entire backlog.